### PR TITLE
Added support for configuring deploy windows in arcanist projects when running `arc land`:

### PR DESCRIPTION
**Pull request by kchen-coursera**
```
// Sample `.arcconfig`
{
  // ...
  "arc.land.validate.deploy_windows.blacklist": [
       "* 16-23 * * FRI",
       "* 0-7 * * MON",
       "* * * * SAT",
       "* * * * SUN",
       "* * 25 12 *"
     ],
  "arc.land.validate.deploy_windows.strategy": "warn"
}

### CODE REVIEW

**Comment by mustafa-coursera** [comment]
we have removed all dependencies from the forked version of arcanist so its easier to update to the FB version. i would strongly recommend extending arcanist and writing a separate workflow that ties into arc land instead of modifying arc land directly

**Comment by kchen-coursera** [comment]
> we have removed all dependencies from the forked version of arcanist so its easier to update to the FB version. i would strongly recommend extending arcanist and writing a separate workflow that ties into arc land instead of modifying arc land directly

I was looking at that but to my understanding arcanist doesn't support extending workflows. The defined extension points were linter, library extension, configuration, and creating new workflows. At the moment, there doesn't appear to be a way to hook into an existing workflow

File: src/workflow/ArcanistLandWorkflow.php [hunk 1]
```diff
@@ -1234,6 +1256,44 @@ private function merge() {
...
   }
 
+  private function handleDeployWinwdows() {
+    // 1. Is the current time in a blacklisted window?
+    $isInFreezeWindow = $this->isInDeployWindows($this->deployWindowsBlacklist, time());
+
+    /**
+     * If in window, depending on the strategy:
+     *  strategy=warn: ask user to confirm or exit
+     *  strategy=reject: simply reject
```
Comment (dguo-coursera) [inline src/workflow/ArcanistLandWorkflow.php]: I'd like to push back on even allowing `strategy=reject` as an option. Seems rather draconian. For our purposes, warn seems sufficient.

Comment (kchen-coursera) [reply src/workflow/ArcanistLandWorkflow.php]: That's fair. I removed it for now

**Pull request merged**